src/data/mv_polynomial.lean

```diff
@@ -129,7 +129,12 @@ local attribute [instance] coeff_coe_to_fun
 def monomial (s : σ →₀ ℕ) (a : α) : mv_polynomial σ α := single s a
 
 /-- `C a` is the constant polynomial with value `a` -/
-def C (a : α) : mv_polynomial σ α := monomial 0 a
+def C : α →+* mv_polynomial σ α :=
```

Comment:
How about making `monomial` and `add_monoid_hom` first?

Reply:
Yes, I agree.

Could I ask though that I defer making significant further changes for a later PR? I definitely want to change a lot more in `polynomial.lean` and `mv_polynomial.lean`, but the `cayley_hamilton` branch can go in for review once this PR is merged, and if we wait for a thorough refactor of polynomials it will start rotting. :-)

Reply:
Sure!